st2common/st2common/constants/meta.py

```diff
@@ -14,12 +14,24 @@
 # limitations under the License.
 
 from __future__ import absolute_import
+
 import yaml
 
+try:
+    from yaml import CSafeLoader as YamlSafeLoader
+except ImportError:
+    from yaml import SafeLoader as YamlSafeLoader
+
 __all__ = [
     'ALLOWED_EXTS',
     'PARSER_FUNCS'
 ]
 
+
+# NOTE: We utilize CSafeLoader if available since it uses C extensions and is faster.
+def yaml_safe_load(stream):
+    return yaml.load(stream, Loader=YamlSafeLoader)
```

Comment:
Using yaml.load will raise an alert when we run security scan / static analysis on st2 codebase because yaml.load is not safe to load text from untrusted source.

Reply:
@m4dcoder There is no security concern  here - it's using ``YamlSafeLoader`` (which is either a native Python implementation or C extension one).

And yes, I'm aware of various static analysis tools - they all have limitations and usually detect many false positives.

This line represents no security risks since it uses ``YamlSafeLoader`` which is the same thing used internally by ``yaml.safe_load()`` - but we can't use it directly since we want to use faster C implementation.

I'm not saying that security analysis tools are bad, but results should always be taken with grain of salt and manually audited for false positives, etc. (same as the openapi spec one - even though we didn't use safe load there it didn't represent security risk since it loaded a static file which is not controlled by the end user - but yeah, I still agree that we should use safe load there just to be on the safe side).

Reply:
If you want, I can add a code comment and make it explicit, but I kinda assumed it wasn't necessary since it's using SafeLoader class (and as pointed in orquests PR, there are even tests for that scenario).

Reply:
Yeah, documenting that in a comment would be good for the next person who investigates this issue.

Is there a way to turn off that security check for that one line? This might be one of those rare times that that is acceptable.

Reply:
Will add a comment.

As far as turning off the check -goes  yeah, that will need to be handled on the tool side (aka nothing StackStorm/st2 specific).

Technically, if it's some commonly used (open source) tool, I would even be fine with adding some tool specific pragma to that line itself, but otherwise it will need to be handled on the tool side - most such tools have internal database for marking and handling false positives, etc.

And honestly, if that was the only false positive returned by this tool on StackStorm/st2 codebase, I would be vastly surprised :)

Over the years, I managed and set up many such tools and there were always, tons and tons of false positives, even with more advanced tools which use AST based rules + ML (I would assume this one does very simple string / regex based searches for common vulnerabilities based on the pattern above, but I could also be wrong).

Reply:
Added a comment in 167ca3f2a91cd20f38b32ab204855a1e86785c67.

Reply:
I reviewed pyyaml and safe_load is just calling load but pass in SafeLoader which is the same as what we coded here. I think it's good to put comment describing its use with SafeLoader and that it's the same as calling safe_load. This will help the person who is going to touch this code next and unintentionally revert to default loader. If I didn't reviewed the pyyaml code, a casual contributor would not know that this is not insecure use because it's not using safe_load. The static analysis tool has limitation I agree. In those cases, we need to document while this is a false positive when we identified it so we don't keep circling back to this. Thanks.